### ISSUE TITLE
Use build args + customization

### ISSUE THREAD

**Issue by sem4phor** [BlackPepperSoftware/docker-craftcms#9]
Hello! (And sorry for opening another issue)

I wanted to set up a basic craft environment with a specified craft version (3.1.12) and some Customizations like described in the customization section of the README.

So a pretty similar setup to the provided demo with following Docker settings:

compose: 

```
version: '3'

services:
  site:
    build:
      context: .
      args:
        CRAFT_VERSION: 3.1
        CRAFT_BUILD: 12
    environment:
      DB_SERVER: database
      DB_USER: craft
      DB_PASSWORD: password
      DB_DATABASE: craft
      DB_DRIVER: mysql
      CRAFT_SITE_URL: http://localhost:8080/
    links:
      - database
    ports:
      - "8080:80"
  database:
    image: mariadb:10
    environment:
      MYSQL_ROOT_PASSWORD: password
      MYSQL_USER: craft
      MYSQL_PASSWORD: password
      MYSQL_DATABASE: craft
    ports:
      - "3306:3306"
``` 

Dockerfile:
```
ARG CRAFT_VERSION
ARG CRAFT_BUILD
FROM blackpepper/craftcms
COPY public /var/www/web
```

So the difference is that i use build args which i want to pass to the FROM base image of the dockerfile.
In the issue https://github.com/BlackPepperSoftware/docker-craftcms/issues/8 this worked because i didnt have a customized dockerfile instead i linked to your dockerfile directly via the context field in the compose file. SO setting build args worked.

Now i consume the your image via FROM in the dockerfile. This causes the build args from compose not being applied to your image. 
Do you have any idea how to pass/overwrite build args from compose to the FROM base image inside the Dockerfile? 

Kindly regards

**Comment by markhobson** [comment]
Hi, no problem. It's a good question but I don't believe that it's something that's supported by Docker itself. Using a context GitHub URL is a Docker Compose feature only.

Ideally we would upgrade our image to CraftCMS 3.1.12 and you could use the latest. What's stopping us is that 3.1 is a breakpoint release, which I believe requires us to upgrade to 3.0.40.1 first, but until the [archive version mismatches](https://github.com/craftcms/cms/issues/3871) are fixed we can't reliably do that.

Let me make sure that we can't just jump straight to 3.1.12 ourselves. Failing that I'd suggest manually building this image at 3.1.12 and pushing it to your own Docker registry. You can then reference that from your Dockerfile.

**Comment by markhobson** [comment]
I've tried updating our CraftCMS instance from 3.0.37 to 3.1.14 and many of our plugins break. We'll need to wait for the Craft archives to be fixed before we can tackle this migration, and hence update the Craft version in this image.

Are you okay to build and push your own updated Craft image in the meantime?

**Comment by sem4phor** [comment]
Okay, yes i will give your suggestion to build my own image a try :) ty for your time!

**Comment by hdpe** [comment]
Hi @sem4phor - just to let you know we've updated the default Craft version in the image to 3.1.14, so you may now be able to use the latest in Docker Hub instead of your own image.
